students/kegan/session10/MailroomTools.py

```diff
@@ -0,0 +1,489 @@
+            'Dear {donor},\nThank you for your generous gift{s} of ' +\
+            '{first}{and}{rest}. Your donation{s}{totalling} will ' +\
+            'go towards feeding homeless kittens in Seattle. ' +\
+            'From the bottom of our hearts, we at Miuvenile Care thank you.' +\
+            '\n\nRegards,\nBungelina Bigglesnorf\nChairwoman, Miuvenile Care'
+        return message.format(**substrings)
+
+    def add(self, *donations):
+        """ Adds passed donations to this donor.
+        Args:
+            donations (args) : donations as arguments
+        """
+        donations = self.intake_donations(*donations)
+        self._donations.extend(donations)
+
+    def __str__(self):
+        """ Returns this donor as a string.
+        Returns:
+            str : donor as string
+        """
+        donations = ', '.join([
+            '${:,.2f}'.format(d) for d in self.donations])
+        return '{}: {}'.format(self.name, donations)
+
+    def __repr__(self):
+        """ Returns representation of this donor.
+        Returns:
+            str : representation of this donor
+        """
+        donations = ', '.join([
+            str(d) for d in self.donations])
+        return 'Donor("{}", {})'.format(self.name, donations)
+
+    def __eq__(self, other):
+        """ Returns whether this donor is equal to other in
+        both name and all donations.
+        Args:
+            other (Donor) : donor to compare
+        Returns:
+            bool : True if other donor is equal to this donor
+        """
+        return (
+            self.name == other.name and
+            self.donations == other.donations)
+
+    def __lt__(self, other):
+        """ Returns whether this donor is less than
+        other. Compares name and donations.
+        Args:
+            other (Donor) : donor to compare
+        Returns:
+            bool : True if this donor is less than other
+        """
+        return (
+            self.name < other.name and
+            self.donations < other.donations)
+
+    def __contains__(self, donation):
+        """ Returns whether given donation amount
+        has been donated by this donor.
+        Args:
+            donation (int or float) : donation to search for
+        Returns:
+            bool :
+                True if donation is in donation list
+                False otherwise
+        """
+        return donation in self.donations
+
+    def multiply(self, factor, min_donation=None, max_donation=None):
+        donations = map(lambda d: self.apply_factor(
+            d, factor, min_donation, max_donation), self.donations)
+        return Donor(self.name, *donations)
+
+    @staticmethod
+    def apply_factor(d, factor, min_donation, max_donation):
+        within_min = True if min_donation is None else d >= min_donation
+        within_max = True if max_donation is None else d <= max_donation
+        return d * factor if within_min and within_max else d
+
+
+@functools.total_ordering
```

Comment:
this is for implementing total ordering on individual items -- are you ever going to need to compare different DonorLists ?